### PR TITLE
[tf] Refactoring top-level security tests

### PR DESCRIPTION
**Pull request by nmittler**
This refactors all of the top-level security tests to use the
common echo deployment and the new echotest framework. Also moves
the authz tests back to the top-level, since it can now share
the same TestMain.

**Please provide a description of this PR:**

### CODE REVIEW

**Review by GregHanson — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

File: pkg/test/framework/components/jwt/server.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+package jwt
+
+import (
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+// Server for custom authz.
```
Comment (aryan16) [inline pkg/test/framework/components/jwt/server.go:23]: nit : It should be authn right ?

File: pkg/test/framework/components/jwt/server.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+type Server interface {
+	Namespace() namespace.Instance
+	FQDN() string
+	HTTPPort() int
+	HTTPSPort() int
+	JwksURI() string
+}
+
+// New creates a new authz Server.
```
Comment (aryan16) [inline pkg/test/framework/components/jwt/server.go:32]: nit : same as above

**Review by aryan16 — APPROVED** [review]

**Pull request merged**